src/lib.rs

```diff
@@ -4,41 +4,132 @@
+//! as the GIL is held, which is why using them doesn't require a  [`Python<'py>`](crate::Python) token.
+//!  The underlying Python object, if mutable, can be mutated through any reference.
+//!
+//! See the [guide](https://pyo3.rs/main/types.html) for an explanation of the different Python object types.
+//!
+//! ## The GIL-independent types
+//!
+//! When wrapped in [`Py`]`<...>`, like with [`Py`]`<`[`PyAny`](crate::types::PyAny)`>` or [`Py`]`<SomePyClass>`, Python objects
+//! no longer have a limited lifetime which makes them easier to store in structs and pass between functions.
+//! However, you cannot do much with them without a
+//! [`Python<'py>`](crate::Python) token, for which you’d need to reacquire the GIL.
+//!
+//! ## PyErr
+//!
+//! The vast majority of operations in this library will return [`PyResult<...>`](PyResult).
 //! This is an alias for the type `Result<..., PyErr>`.
 //!
-//! A `PyErr` represents a Python exception. Errors within the `PyO3` library are
-//! also exposed as Python exceptions.
+//! A `PyErr` represents a Python exception. A `PyErr` returned to Python code will be raised as a Python exception.
+//! Errors from `PyO3` itself are also exposed as Python exceptions.
+//!
+//! # Feature flags
+//!
+//! PyO3 uses [feature flags](https://doc.rust-lang.org/cargo/reference/features.html)
+//! to enable you to opt-in to additional functionality. For a detailed description, see
+//! the [Features Reference chapter of the guide](https://pyo3.rs/main/features.html#features-reference).
+//!
+//! ## Default feature flags
+//!
+//! The following features are turned on by default:
+//! - `macros`: Enables various macros, including all the attribute macros.
+//!
+//! ## Optional feature flags
+//!
+//! The following features are optional:
+//! - `abi3`: Restricts PyO3's API to a subset of the full Python API which is guaranteed
+//! by [PEP 384](https://www.python.org/dev/peps/pep-0384/) to be forwards-compatible with future Python versions.
+//
+//! - `auto-initialize`: Changes [`Python::with_gil`](crate::Python::with_gil) and
+//! [`Python::acquire_gil`](crate::Python::acquire_gil) to automatically initialize the
+//! Python interpreter if needed.
+//
+//! - `extension-module`: This will tell the linker to keep the Python symbols unresolved,
+//! so that your module can also be used with statically linked Python interpreters.
+//!Use this feature when building an extension module.
+//
+//! - `hashbrown`: Enables conversions between Python objects and
+//! [hashbrown](https://docs.rs/hashbrown)'s
+//! [`HashMap`](https://docs.rs/hashbrown/latest/hashbrown/struct.HashMap.html) and
+//! [`HashSet`](https://docs.rs/hashbrown/latest/hashbrown/struct.HashSet.html) types.
+//
+//! - `multiple-pymethods`: Enables the use of multiple
+//! [`#[pymethods]`](crate::proc_macro::pymethods) blocks per
+//! [`#[pyclass]`](crate::proc_macro::pyclass).
```

Comment:
```suggestion
//! [`#[pyclass]`](crate::proc_macro::pyclass), at the cost of some portability.
```

Reply:
What does cost of portability mean in this context? Does it not work on some platforms? If so, which ones?

Reply:
From the guide: "Most users should only need a single `#[pymethods]` per `#[pyclass]`. In addition, not all platforms (e.g. Wasm) are supported by `inventory`. For this reason this feature is not enabled by default, meaning fewer dependencies and faster compilation for the majority of users."

Reply:
I'll add this:

> This adds a dependency on the [`inventory`](https://docs.rs/inventory) crate, which is not supported on all platforms.